// Copyright (c) 2019 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;
option java_outer_classname = "PayloadProto";

import "bosdyn/api/geometry.proto";
import "bosdyn/api/header.proto";

// A Payload describes a single payload installed on the Spot platform.
// It includes all external information necessary to represent
// the payload to the user as a single record.
message Payload {
    // A unique id provided by the payload or auto-generated by the website.
    string GUID = 1;
    // A human readable name describing this payload. It is provided by the
    // payload as part of the payload announcement system.
    string name = 2;
    // A human-readable description string providing more context as to the
    // function of this payload. It is displayed in UIs.
    string description = 3;
    // A label prefix is used to indicate what type of payload this is.
    repeated string label_prefix = 9;
    // Set true once the payload is authorized by the administrator in the payload webpage.
    bool is_authorized = 4;
    // Set true if the payload is attached to the robot.
    bool is_enabled = 5;
    // Set true for payloads registered without their own computers. These records
    // are all manually entered.
    bool is_noncompute_payload = 6;
    // The pose of the payload relative to the body frame.
    SE3Pose body_T_payload = 7;
    // The pose of the payload relative to the mount frame.
    SE3Pose mount_T_payload = 8;
}

message ListPayloadsRequest {
    RequestHeader header = 1;
}

message ListPayloadsResponse {
    ResponseHeader header = 1;

    repeated Payload payloads = 2;
}
